diff -Narup fuppes-0.660.orig/src/plugins/metadata_libavformat.c fuppes-0.660/src/plugins/metadata_libavformat.c
--- fuppes-0.660.orig/src/plugins/metadata_libavformat.c	2012-11-11 23:33:43.718819733 +0900
+++ fuppes-0.660/src/plugins/metadata_libavformat.c	2012-11-13 18:02:10.809291105 +0900
@@ -59,13 +59,14 @@ void register_fuppes_plugin(plugin_info*
 int fuppes_metadata_file_open(plugin_info* plugin, const char* fileName)
 {
 	plugin->user_data = NULL;
+	AVFormatContext** ctx = (AVFormatContext**)&plugin->user_data;
 	
-	if(av_open_input_file(&plugin->user_data, fileName, NULL, 0, NULL) != 0) {
+	if(avformat_open_input(ctx, fileName, NULL, NULL) != 0) {
 		return -1;
 	}
 		
-	if(av_find_stream_info((AVFormatContext*)plugin->user_data) < 0) {
-		av_close_input_file((AVFormatContext*)plugin->user_data);
+	if(avformat_find_stream_info(*ctx, NULL) < 0) {
+		avformat_close_input(ctx);
 	  return -1;
 	}
 	
@@ -74,6 +75,8 @@ int fuppes_metadata_file_open(plugin_inf
 
 int fuppes_metadata_read(plugin_info* plugin, metadata_t* metadata)
 {
+	AVFormatContext* ctx = (AVFormatContext*)plugin->user_data;
+
 	// duration
 	if(((AVFormatContext*)plugin->user_data)->duration != AV_NOPTS_VALUE) {
  
@@ -149,21 +152,21 @@ int fuppes_metadata_read(plugin_info* pl
 		} */
 
 		switch(pStream->codec->codec_type) {
-			case CODEC_TYPE_VIDEO:
+			case AVMEDIA_TYPE_VIDEO:
 				metadata->type		= MD_VIDEO;
 				metadata->width 	= pStream->codec->width;
 				metadata->height	= pStream->codec->height;
 				set_value(&metadata->video_codec, codec_name);
 				break;
-			case CODEC_TYPE_AUDIO:
+			case AVMEDIA_TYPE_AUDIO:
 				set_value(&metadata->audio_codec, codec_name);
 				if(metadata->type == MD_NONE) {
 					metadata->type = MD_AUDIO;
 				}
 				break;
-			case CODEC_TYPE_DATA:
+			case AVMEDIA_TYPE_DATA:
 				break;
-			case CODEC_TYPE_SUBTITLE:
+			case AVMEDIA_TYPE_SUBTITLE:
 				break;
 			default:
 				break;
