diff -Narup fuppes-0.660.orig/src/lib/Common/Common.cpp fuppes-0.660/src/lib/Common/Common.cpp
--- fuppes-0.660.orig/src/lib/Common/Common.cpp	2012-11-11 23:33:43.768828873 +0900
+++ fuppes-0.660/src/lib/Common/Common.cpp	2012-11-12 19:08:56.368611781 +0900
@@ -35,6 +35,7 @@
 #include "../SharedConfig.h"
 
 #include <cstdio>
+#include <unistd.h>
 
 #ifndef WIN32
 #include <dirent.h>
diff -Narup fuppes-0.660.orig/src/plugins/database_sqlite3.cpp fuppes-0.660/src/plugins/database_sqlite3.cpp
--- fuppes-0.660.orig/src/plugins/database_sqlite3.cpp	2012-11-11 23:33:43.735489447 +0900
+++ fuppes-0.660/src/plugins/database_sqlite3.cpp	2012-11-12 19:08:56.385281495 +0900
@@ -29,6 +29,7 @@
 #include <string>
 #include <sstream>
 #include <iostream>
+#include <unistd.h>
 
 #include <sqlite3.h>
 
diff -Narup fuppes-0.660.orig/src/plugins/metadata_ffmpegthumbnailer.cpp fuppes-0.660/src/plugins/metadata_ffmpegthumbnailer.cpp
--- fuppes-0.660.orig/src/plugins/metadata_ffmpegthumbnailer.cpp	2012-11-11 23:33:43.628803281 +0900
+++ fuppes-0.660/src/plugins/metadata_ffmpegthumbnailer.cpp	2012-11-12 19:08:56.391949381 +0900
@@ -26,8 +26,8 @@
 
 #include "../../include/fuppes_plugin.h"
 
-#include <videothumbnailer.h>
-#include <imagetypes.h>
+#include <libffmpegthumbnailer/videothumbnailer.h>
+#include <libffmpegthumbnailer/imagetypes.h>
 
 #include <vector>
 #include <iostream>
